### DISCUSSION TITLE
server part - event.context

### DISCUSSION THREAD

**Discussion by borodadada** [nuxt/nuxt#17344] [Questions] [answered]
server/middleware/test.ts

```
export default defineEventHandler((event) => {
    event.context.myVariable = 123
})
```
Hi! How i can get this variable in my app vue?

**Comment by Flanker72** [comment, marked as answer]
Seems you are trying to mix presentation part (Vue) and server-side logic (API).
To use some server data within your Vue application, you need to directly pass it from the backend (call $fetch ou useFetch/useAsyncData to load the results).

This king of middleware (server/middleware) is designed to work with server APIs and server pages, they both have separate logic than Vue application.

Server folder is designed to make full-stack application, including DB access, API exposing, etc.

Event context is designed to pass data between middlewares and APIs. Since API calls are stateless, you need to have a some kind of data accessible between middleware calls.

So, if you need to pass any data from server middleware to your client application, try to use data fetching, cookies or some other approach.

**Reply by borodadada** [reply]
thx man
